src/receivers/HTTPReceiver.ts

```diff
@@ -302,19 +314,35 @@ export default class HTTPReceiver implements Receiver {
       // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
       const [installPath, installRedirectUriPath] = [this.installPath!, this.installRedirectUriPath!];
 
+      // Visiting the installation endpoint
       if (path === installPath) {
         // Render installation path (containing Add to Slack button)
         return this.handleInstallPathRequest(res);
       }
+
+      // Installation has been initiated
       if (path === installRedirectUriPath) {
         // Handle OAuth callback request (to exchange authorization grant for a new access token)
         return this.handleInstallRedirectRequest(req, res);
       }
     }
 
+    // Handle custom routes
+    if (this.customRoutes.length) {
+      const match = this.customRoutes.find((route) => {
+        const isMethodMatch = Array.isArray(route.method) ?
+          route.method.includes(method) : route.method === method;
+        return route.path === path && isMethodMatch;
+      });
+
+      if (match) {
+        return match.callback(req, res);
+      }
+    }
+
     // If the request did not match the previous conditions, an error is thrown. The error can be caught by the
     // the caller in order to defer to other routing logic (similar to calling `next()` in connect middleware).
-    throw new HTTPReceiverDeferredRequestError('Unhandled HTTP request', req, res);
+    throw new HTTPReceiverDeferredRequestError(`Unhandled HTTP request made to ${path}`, req, res);
```

Comment:
Should we also log out the HTTP verb here, since that is a matching parameter too?

Reply:
Great idea! Will adjust to include it.